pkg/management/component.go

```diff
@@ -785,18 +790,25 @@ func (c *component) GetRolesOfUser(ctx context.Context, realmName, userID string
 		c.logger.Warn(ctx, "err", err.Error())
 		return nil, err
 	}
+	userRoles, err := c.getRolesWithAttributes(accessToken, realmName, rolesKc)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Failed to get user roles", "realm", realmName, "user")
+		return nil, err
+	}
 
 	var rolesRep = []api.RoleRepresentation{}
-	for _, roleKc := range rolesKc {
-		var roleRep api.RoleRepresentation
-		roleRep.ID = roleKc.ID
-		roleRep.Name = roleKc.Name
-		roleRep.Composite = roleKc.Composite
-		roleRep.ClientRole = roleKc.ClientRole
-		roleRep.ContainerID = roleKc.ContainerID
-		roleRep.Description = roleKc.Description
-
-		rolesRep = append(rolesRep, roleRep)
+	for _, roleKc := range userRoles {
+		if c.isBusinessRole(roleKc) {
```

Comment:
Factorize? You can write convert functions in api/management

Reply:
Good catch ! Done !